### PR TITLE
Add example of using `sunpy.net.attr.or_`

### LINKED ISSUE #6371: Document `sunpy.net.attr.or_` somewhere
### Describe the feature

The hidden feature of `sunpy.net.attr.or_` could be useful to some, should be on show somewhere

e.g. 

```python
wavelengths = [94*u.angstrom, 131*u.angstrom, 171*u.angstrom, 193*u.angstrom, 211*u.angstrom]
aia_search= Fido.search(a.Time("2022-02-15 21:40", "2022-02-15 23:00"), 
                        a.Instrument("AIA"), 
                        attr.or_(*[a.Wavelength(wav) for wav in wavelengths]))

```
### Proposed solution

add it to docs

### PR DESCRIPTION
**Pull request by hayesla**
## PR Description

This adds an example of how to search for multiple wavelengths with Fido, fixes #6371

### CODE REVIEW

**Comment by Cadair** [comment]
This looks good, I have posted a question about making this API nicer in the original issue.

**Comment by wtbarnes** [comment]
Could you also add an example of using the `|` operator so that we have all three approaches (one wavelength, `|`, `or_`) in one place?

Thanks for doing this. I think this example also nicely follows the "How to..." format.

**Comment by hayesla** [comment]
thats a nice idea, I'll do that now

File: changelog/6501.doc.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+Add example of how to search for multiple wavelengths attributes for AIA data using sunpy.net.attr.AttrOr.
```
Comment (nabobalis) [inline changelog/6501.doc.rst:1]:
```suggestion
Added an example (:ref:`sphx_glr_generated_gallery_acquiring_searching_multiple_wavelengths.py`) of how to search for multiple wavelengths attributes for AIA data using `sunpy.net.attr.AttrOr`.
```
Hopefully this works.

File: examples/acquiring_data/searching_multiple_wavelengths.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
+"""
+==============================================
+Searching for multiple wavelengths with Fido
+==============================================
+This example shows how you can search for several wavelengths
+of AIA data with Fido.
```
Comment (nabobalis) [inline examples/acquiring_data/searching_multiple_wavelengths.py:6]:
```suggestion

This example shows how you can search for several wavelengths
of AIA data with Fido.
```

File: examples/acquiring_data/searching_multiple_wavelengths.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+"""
+from astropy import units as u
+
+from sunpy.net import Fido, attr
+from sunpy.net import attrs as a
+
+###############################################################################
+# Here we are demonstrating how you can search for specific wavelengths of
+# AIA data using `Fido` and the `sunpy.net.attr.AttrOr` function.
```
Comment (nabobalis) [inline examples/acquiring_data/searching_multiple_wavelengths.py:15]:
```suggestion
# AIA data using `Fido <sunpy.net.fido_factory.UnifiedDownloaderFactory>` and the `sunpy.net.attr.AttrOr` function.
```
Does the link to Fido work? I had always assumed not.

File: examples/acquiring_data/searching_multiple_wavelengths.py [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+# For example, you may only want a single wavelength, say 171 Angstrom.:
```
Comment (nabobalis) [inline examples/acquiring_data/searching_multiple_wavelengths.py:16]:
```suggestion
# For example, you may only want a single wavelength, say 171 Angstrom:
```

File: examples/acquiring_data/searching_multiple_wavelengths.py [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+aia_search = Fido.search(a.Time("2022-02-20 00:00", "2022-02-20 00:01"),
+                         a.Instrument("AIA"),
+                         a.Wavelength(171*u.angstrom))
+
+aia_search
```
Comment (nabobalis) [inline examples/acquiring_data/searching_multiple_wavelengths.py:22]:
```suggestion
print(aia_search)
```
I think this needs to be printed to ensure it always renders.

File: examples/acquiring_data/searching_multiple_wavelengths.py [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+# But say you actually want to search for several wavelengths, rather than just one.
+# You could use the "|" operator, or instead you can use the `sunpy.net.attr.AttrOr`
+# function.
+
+wavelengths = [94, 131, 171, 193, 211]*u.angstrom
+aia_search = Fido.search(a.Time("2022-02-20 00:00", "2022-02-20 00:01"),
+                         a.Instrument("AIA"),
+                         attr.AttrOr([a.Wavelength(wav) for wav in wavelengths]))
+
+aia_search
```
Comment (nabobalis) [inline examples/acquiring_data/searching_multiple_wavelengths.py:34]:
```suggestion
print(aia_search)
```

File: examples/acquiring_data/searching_multiple_wavelengths.py [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+
+# This returns several searches for each of the wavelengths, which can be indexed.
+# Here the first index is that of 94 angstrom.
+aia_search[0]
```
Comment (nabobalis) [inline examples/acquiring_data/searching_multiple_wavelengths.py:38]:
```suggestion
print(aia_search[0])
```

File: examples/acquiring_data/searching_multiple_wavelengths.py [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+
+###############################################################################
+# You can then pass the Fido result to Fido.fetch to download the data.
+# i.e. Fido.fetch(aia_search)
```
Comment (nabobalis) [inline examples/acquiring_data/searching_multiple_wavelengths.py:42]:
```suggestion
# i.e., ``Fido.fetch(aia_search)``
```

File: examples/acquiring_data/searching_multiple_wavelengths.py [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+                         attr.AttrOr([a.Wavelength(wav) for wav in wavelengths]))
+
+aia_search
+
+# This returns several searches for each of the wavelengths, which can be indexed.
+# Here the first index is that of 94 angstrom.
+aia_search[0]
+
+###############################################################################
+# You can then pass the Fido result to Fido.fetch to download the data.
```
Comment (nabobalis) [inline examples/acquiring_data/searching_multiple_wavelengths.py:41]:
```suggestion
# You can then pass the result to ``Fido.fetch`` to download the data.
```

File: examples/acquiring_data/searching_multiple_wavelengths.py [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+
+###############################################################################
+# But say you actually want to search for several wavelengths, rather than just one.
+# You could use the "|" operator, or instead you can use the `sunpy.net.attr.AttrOr`
+# function.
+
+wavelengths = [94, 131, 171, 193, 211]*u.angstrom
+aia_search = Fido.search(a.Time("2022-02-20 00:00", "2022-02-20 00:01"),
+                         a.Instrument("AIA"),
+                         attr.AttrOr([a.Wavelength(wav) for wav in wavelengths]))
```
Comment (Cadair) [inline examples/acquiring_data/searching_multiple_wavelengths.py:32]: If we are going to start documenting this should we not pull `AttrOr` and `AttrAnd` into the `a.` namespace?

Comment (nabobalis) [reply examples/acquiring_data/searching_multiple_wavelengths.py:32]: If we plan to start using it like this, 100%

Comment (dstansby) [reply examples/acquiring_data/searching_multiple_wavelengths.py:32]: I opened a new issue for this at https://github.com/sunpy/sunpy/issues/6575

**Review by nabobalis — APPROVED** [review]

**Pull request merged**